pkg/nuctl/test/function_test.go

```diff
@@ -681,6 +684,122 @@ func (suite *functionDeployTestSuite) TestBuildAndDeployFromFileWithOverriddenAr
 	suite.Require().Contains(suite.outputBuffer.String(), randomString)
 }
 
+func (suite *functionDeployTestSuite) TestDeployWithResourceVersion() {
+
+	// TODO: when enable some sort of resource validation on other platforms, remove platform ensuring
+	suite.ensureRunningOnPlatform("kube")
+
+	// read and parse the function we're gonna test
+	functionConfig := functionconfig.Config{}
+	uniqueSuffix := "-" + xid.New().String()
+	functionPath := path.Join(suite.GetFunctionsDir(), "common", "json-parser-with-function-config", "python")
+	functionBody, err := ioutil.ReadFile(filepath.Join(functionPath, build.FunctionConfigFileName))
+	suite.Require().NoError(err)
+	err = yaml.Unmarshal(functionBody, &functionConfig)
+	suite.Require().NoError(err)
+
+	// name uniqueness
+	functionConfig.Meta.Name = functionConfig.Meta.Name + uniqueSuffix
+
+	// ensure no resource version
+	functionConfig.Meta.ResourceVersion = ""
+
+	// --- step 1 ---
+	// deploy first time, should successfully create the function
+
+	// write function config to file
+	functionConfigPath, err := suite.writeFunctionConfigToTemp(&functionConfig,
+		"resource-version-*.yaml")
+	suite.Require().NoError(err)
+
+	// remove when done
+	defer os.RemoveAll(functionConfigPath)
```

Comment:
```
	defer os.RemoveAll(functionConfigPath)
```
Appears 3 times.
I believe you need to use unique variable for each.
Explanation:
Each time there's a different path `functionConfigPath`.
At the end of the test only 2 `os.RemoveAll()` will be called but all with the latest contents of `functionConfigPath`, 2 paths will remain unremoved